tiledb/sm/array_schema/dimension.cc

```diff
@@ -1263,6 +1310,27 @@ uint64_t Dimension::map_to_uint64_3(
   return (norm_coord_T / dom_range_T) * max_bucket_val;
 }
 
+template <>
+uint64_t Dimension::map_to_uint64_3<double>(
```

Comment:
Don't we need a similar one for `float`?

Reply:
@stavros for floats we are ok here since we cast to double for the arithmetics.